### PR TITLE
Define REST apis for Environment model

### PR DESCRIPTION
**Pull request by maheshkelkar**
References #59

### CODE REVIEW

File: features/api/environment/reading.feature [hunk 1]
```diff
@@ -0,0 +1,98 @@
+Feature: Environment READ APIs
+
+  As a RESTful client or user
+  I should be able to read information about environments
+
+  @wip
+  Scenario: Fetching all environments 
+
+    Given a integ environment configuration:
```
Comment (rtyler) [inline features/api/environment/reading.feature]: "integ" means nothing as far as DeployDB is concerned, I suggest changing this to:

```cucumber
Given an environment configuration:
```

File: features/api/environment/reading.feature [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    description: "DeployDB Primary Integration"
+    webhooks:
+      deployment:
+        created:
+          - http://jenkins.example.com/job/integ-deploy-created/build
+        completed:
+          - http://jenkins.example.com/job/integ-deploy-completed/build
+    """
+    And another dev-alpha environment configuration:
```
Comment (rtyler) [inline features/api/environment/reading.feature]: I don't think this "and" step is necessuary, you should remove it

File: features/api/environment/reading.feature [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    webhooks:
+      deployment:
+        created:
+          - http://jenkins.example.com/job/dev-alpha-deploy-created/build
+        completed:
+          - http://jenkins.example.com/job/dev-alpha-deploy-completed/build
+    """
+    When I GET "/api/environments"
+    Then the response should be 200
+    And the body should be JSON:
```
Comment (rtyler) [inline features/api/environment/reading.feature]: :+1:

File: features/api/environment/reading.feature [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                "started" : [],
+                "created" : ["http://jenkins.example.com/job/dev-alpha-deploy-created/build"],
+                "completed" : ["http://jenkins.example.com/job/dev-alpha-deploy-completed/build"]
+            },
+            "promotions" : {
+                "completed" : []
+            }
+        }
+      }]
+    """
```
Comment (rtyler) [inline features/api/environment/reading.feature]: There should be two new lines separating scenarios

File: features/api/environment/reading.feature [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                "started" : [],
+                "created" : ["http://jenkins.example.com/job/integ-deploy-created/build"],
+                "completed" : ["http://jenkins.example.com/job/integ-deploy-completed/build"]
+            },
+            "promotions" : {
+                "completed" : []
+            }
+        }
+      }
+    """
```
Comment (rtyler) [inline features/api/environment/reading.feature]: 2 * \n

File: features/api/environment/reading.feature (hunk 2 above)
Comment (maheshkelkar) [inline features/api/environment/reading.feature]: Without that "And", we need to manipulate strings in order to split and save into 2 seperate files.

Comment (rtyler) [inline features/api/environment/reading.feature]: I'm suggesting only putting one environment in the response below, I don't know what value we get from having both

Comment (maheshkelkar) [inline features/api/environment/reading.feature]: We are testing getAll() functionality; also "And" is not necessary. I can repeat the "Given" twice.

Comment (rtyler) [inline features/api/environment/reading.feature]: What I'm suggesting is that it's not something that's useful from a documentation standpoint. Sure in the Spock tests we should verify this much more, but an Array respoinse with one element I think is fine for the cukes

**Comment by maheshkelkar** [comment]
@rtyler @GiriDandu I am incorporated all the comments. I think its good to go.

**Comment by rtyler** [comment]
Woohoo! :+1:

**Pull request merged**